analyzers/tests/SonarAnalyzer.Test/TestCases/LoopsAndLinq.CSharp11.cs

```diff
@@ -144,3 +145,24 @@ void Test(IEnumerable<ReadOnlyMemory<char>> tokens)
         }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/8566
+class Repro_8566
+{
+    IAsyncEnumerable<object> Bar() => throw new NotImplementedException();
+
+    bool Test(object item) => throw new NotImplementedException();
+
+    public async Task<int> ForEach_IAsyncEnumerable()
+    {
+        int count = 0;
+        await foreach (var item in Bar()) // Noncompliant FP
```

Comment:
I don't consider this an FP. The `foreach` statement can be simplified by using LINQ in this case. `IAsyncEnumerable` has support for LINQ operations if you install [the System.Linq.Async package from Nuget](https://www.nuget.org/packages/System.Linq.Async).
So the analyzer is correct in raising a warning for this code. The problem is the codefix. It converts the code to this (e.g. in call form):
```cs
public async Task<int> Foo()
{
    return Bar().Where(item => Test(item)).Count();
}
```
This is incorrect and doesn't compile. The correct version is:
```cs
public ValueTask<int> Foo()
{
    return Bar().Where(item => Test(item)).CountAsync();
}
```
Or the same with `async/await`.

AFAIK the codefix isn't ours, it's coming from Microsoft (please double-check it, I couldn't find a codefix for this rule in this repo).
Summar summarum: the analyzer behavior looks correct, and we can't repair a codefix that isn't ours, so the issue should be closed.

Reply:
That's great! I was not aware of the `System.Linq.Async` Nuget package. As you mentioned, the codefix should be on Microsoft side. I'll let the user know about this and close this PR and the related issue.